### PR TITLE
Include additional fields needed for traceability in precompile ContractFunctionResult #2933 

### LINKED ISSUE #2933: Enable contract function results for traceability
### Problem

Currently, this feature is disabled. To enable this feature, protobuf and services changes are required.

### Solution

Add missing protobufs and update services

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by dimitar-dinev**
**Description**:
This PR modifies `HTSPrecompiledContract` to include additional fields in the `ContractFunctionResult` of its transaction records in order to support precompile traceability from the mirror nodes.
* `gas`, `amount` and `functionParameters` fields have been added to `EvmFnResult` 
  - `gas` specifies the amount of gas available for the call
  - `amount` specifies the number of tinybars sent.  **NOTE** that currently EVM value transfers are not allowed.
  - `functionParameters` specifies the called called function and the parameters passed into the contract call
* Precompile calls to `HTSPrecompiledContract` now include the three fields in their transaction records whenever their corresponding transaction body is not of type `ContractCallTransactionBody` or `ContractCreateTransactionBody` (currently, this applies to all precompiles, except for those extending the `ERCReadOnlyAbstractPrecompile`

**Related issue(s)**:

Fixes #2933 

**Notes for reviewer**:
The E2E framework extension here is not complete. Extending it to support all precompile functions and adding their corresponding assertions to the rest of the precompile suites should be completed as part of this [issue](https://github.com/LimeChain/hedera-services/issues/830).

### CODE REVIEW

**Review by stoqnkpL — COMMENTED** [review]
Looks great to me! Only added one comment for the shouldEnrich... boolean

File: test-clients/src/main/java/com/hedera/services/bdd/suites/utils/contracts/FunctionParameters.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	public Bytes forMintToken(final byte[] address, final long amount, final List<String> metadata) {
+		this.functionHash = ABI_ID_MINT_TOKEN;
+		final var result = Tuple.of(
+				convertBesuAddressToHeadlongAddress(address),
+				BigInteger.valueOf(amount),
+				metadata.stream().map(String::getBytes).toArray(byte[][]::new)
+		);
+		this.functionParams = Bytes.wrap(mintTokenType.encode(result).array());
+		return getBytes();
+	}
```
Comment (IvanKavaldzhiev) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/utils/contracts/FunctionParameters.java:54]: I think if we want to also encode function parameters for other functions in the future we can apply a builder approach as in [here](https://github.com/hashgraph/hedera-services/blob/ae3db581654ac5a272166e0864bf093776e41a0c/hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/EncodingFacade.java#L167) but make the builder class public. In this way we will have several benefits:

- We can have fields with default values (e.g. final `long amount = 0L`). In this case when we want to construct function parameters for non-fungible mint we just can skip setting the amount like this:

```
functionParametersBuilder()
				.forFunction(FunctionType.MINT)
				.withAddress(token)
				.withMetadata(metadata)
				.build();
```
				
and vice versa, for fungible mint we can have:

```
functionParametersBuilder()
				.forFunction(FunctionType.MINT)
				.withAddress(token)
				.withAmount(amount)
				.build();
```
				
 - We will have common class fields for different operations (e.g. amount - used in burn, mint, transfer, etc.)

**Review by IvanKavaldzhiev — COMMENTED** [review]
It looks great! I just have a proposal for the FunctionParameters encoder.

**Review by stoqnkpL — APPROVED** [review]

File: hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java [hunk 2]
```diff
@@ -547,7 +548,11 @@ private void addContractCallResultToRecord(
 					Collections.emptyList(),
 					Collections.emptyList(),
 					EvmFnResult.EMPTY,
-					Collections.emptyMap());
+					Collections.emptyMap(),
+					precompile.shouldAddTraceabilityFieldsToRecord() ? messageFrame.getRemainingGas().toLong() : 0L,
```
Comment (shemnon) [inline hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java:552]: I think we should create variables for these three fields earlier in the code, at line 542.  That way we check the feature flag only once.

File: pom.xml [hunk 3]
```diff
@@ -122,7 +122,7 @@
 			</snapshots>
 			<id>proto-staging</id>
 			<name>0.24.x Protobufs</name>
-			<url>https://oss.sonatype.org/content/repositories/comhederahashgraph-1410</url>
+			<url>https://oss.sonatype.org/content/repositories/comhederahashgraph-1418</url>
```
Comment (shemnon) [inline pom.xml:125]: This file will need to be adjusted to production versions prior to merge.

**Review by shemnon — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (tinker-michaelj) [reply pom.xml:125]: We have tended to accept `-alpha.x` proto artifacts in Services `master` b/c it's so difficult to truly front-load all the proto changes targeted for a release. 😞

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Pull request merged**